sonar-plugin-api/src/main/java/org/sonar/api/internal/SonarRuntimeFactory.java

```diff
@@ -23,26 +23,29 @@
 import java.io.IOException;
 import java.net.URL;
 import java.nio.charset.StandardCharsets;
+import javax.annotation.Nullable;
+import org.sonar.api.SonarProduct;
+import org.sonar.api.SonarQubeSide;
 import org.sonar.api.SonarQubeVersion;
 import org.sonar.api.utils.System2;
 import org.sonar.api.utils.Version;
 
 /**
  * For internal use only.
  */
-public class SonarQubeVersionFactory {
+public class SonarRuntimeFactory {
 
   private static final String FILE_PATH = "/sq-version.txt";
 
-  private SonarQubeVersionFactory() {
+  private SonarRuntimeFactory() {
     // prevents instantiation
   }
 
-  public static SonarQubeVersion create(System2 system) {
+  public static SonarQubeVersion create(System2 system, SonarProduct product, @Nullable SonarQubeSide sonarQubeSide) {
     try {
       URL url = system.getResource(FILE_PATH);
       String versionInFile = Resources.toString(url, StandardCharsets.UTF_8);
-      return new SonarQubeVersion(Version.parse(versionInFile));
+      return new SonarQubeVersion(Version.parse(versionInFile), product, sonarQubeSide);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "SonarQubeVersion"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)